src/System/Console/Hawk/Args/Parse.hs

```diff
@@ -109,16 +116,33 @@ inputSpec (rSep, fSep) = InputSpec <$> source <*> format
 -- >>> test ["-o\t", "-d,", "-O|"]
 -- UseStdout
 -- ("|","\t")
+--
+-- TODO: write test cases for in-place edit
 outputSpec :: (Functor m, Monad m)
-           => CommonSeparators -> OptionParserT HawkOption m OutputSpec
-outputSpec (r, f) = OutputSpec <$> sink <*> format
+           => CommonOptions -> OptionParserT HawkOption m OutputSpec
+outputSpec options = OutputSpec <$> sink <*> format
   where
-    sink = return UseStdout
-    format = OutputFormat <$> record <*> field
-    record = consumeLast Option.OutputRecordDelimiter r' Option.consumeDelimiter
-    field = consumeLast Option.OutputFieldDelimiter f' Option.consumeDelimiter
-    r' = fromSeparator r
-    f' = fromSeparator f
+    sink = do
+        -- TODO: refactor with upcoming consumeNullable changes
+        backupSuffix <- consumeLast Option.InPlaceEdit "" $
+                            consumeNullable "<none>" consumeString
+        if isNothing file
+            then if null backupSuffix
+                then return UseStdout
+                else fail "in-place edit requires input file"
+            else if null backupSuffix
+                then return UseStdout
+                else if backupSuffix == "<none>"
+                    then return $ OutputFile file' Nothing
+                    else return $ OutputFile file' $ Just $ file' ++ backupSuffix
+
+    format = OutputFormat <$> record' <*> field'
+    record' = consumeLast Option.OutputRecordDelimiter r' Option.consumeDelimiter
+    field' = consumeLast Option.OutputFieldDelimiter f' Option.consumeDelimiter
+    r' = fromSeparator $ record options
+    f' = fromSeparator $ field options
+    file = inputFile options
+    file' = (fromJust file)
```

Comment:
Partial functions like `fromJust` are needlessly error-prone, as the compiler won't be able to catch your mistake if you use `file'` outside of the `else` branch of your `if isNothing file` check. Please pattern-match on the `Maybe` instead:

    case inputFile options of
      Nothing -> ...
      Just file -> ... OutputFile file $ Just $ file ++ backupSuffix


could you instead replace your `if isNothing file then ... else ...` with a `case file of {pattern-match on `file` instead of using `isNothing`?